File: src/main/java/com/severett/kotlinintro/model/Dog.java
package com.severett.kotlinintro.model;

import jakarta.persistence.Entity;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@NoArgsConstructor
@Getter
@Setter
public class Dog extends Pet {
    private static final int HASHCODE = Dog.class.hashCode();

    private double tailLength;

    public Dog(int id, String name, double tailLength) {
        super(id, name);
        this.tailLength = tailLength;
    }

    @Override
    public String getSound() {
        return "Barks";
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Dog)) return false;
        return getId() == ((Dog) o).getId();
    }

    @Override
    public int hashCode() {
        return HASHCODE;
    }
}

File: src/main/kotlin/com/severett/kotlinintro/model/Dog.kt
package com.severett.kotlinintro.model

import jakarta.persistence.Entity

private val HASHCODE = Dog::javaClass.hashCode()

@Entity
class Dog(id: Int = 0, name: String = "", var tailLength: Double = 0.0) : Pet(id, name) {
    override val sound: String
        get() = "Barks"

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Dog) return false
        return id == other.id
    }

    override fun hashCode() = HASHCODE
}
